components/ModalDeleteButton.js

```diff
@@ -0,0 +1,98 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import ReactDOM from "react-dom";
+import styled from "styled-components";
+
+export default function ModalDeleteButton({
```

Comment:
This is not just a Button, but a full Modal. But please try to make the Modal functionality and the __content__ of the modal independent from another. If you struggle with it write a question into the discussions board.

Reply:
I mean by that that you have one generic component called Modal, which accepts children, and then you nest inside it the components which provide the functionality you want.